src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -720,12 +720,19 @@ private static IEdmTypeReference ValidateIsOfOrCast(BindingState state, bool isC
                     Error.Format(SRResources.MetadataBinder_CastOrIsOfExpressionWithWrongNumberOfOperands, args.Count));
             }
 
-            ConstantNode typeArgument = args.Last() as ConstantNode;
+            QueryNode queryNode = args.Last();

```

Comment:
Given the `args.Count` above this line, that suggests to me that we're dealing with a `List<T>`. I think we can avoid using LINQ here